commands/checkout.go

```diff
@@ -74,6 +82,11 @@ func transformCheckoutArgs(args *Args) error {
 		return err
 	}
 
+	if idx := args.IndexOfParam("--no-branch"); idx >= 0 {
+		args.RemoveParam(idx)
+		flagCheckoutNoBranch = true
```

Comment:
This line should not be needed, since `flagCheckoutNoBranch` value should be already updated by the flag parsing mechanism.